pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixServerStarter.java

```diff
@@ -154,6 +160,45 @@ public HelixServerStarter(String helixClusterName, String zkAddress, PinotConfig
             Server.DEFAULT_CURRENT_DATA_TABLE_VERSION));
   }
 
+  /**
+   *  Invoke pinot environment provider factory's init method to register the environment provider &
+   *  fetch the overridden server configs for the invoked environment provider.
+   */
+  private void fetchEnvironmentSpecificConfigs() {
+    PinotConfiguration environmentProviderConfigs = _serverConf.subset(
+        Server.PREFIX_OF_CONFIG_OF_ENVIRONMENT_PROVIDER_FACTORY);
+
+    if (environmentProviderConfigs.toMap().isEmpty()) {
+      LOGGER.info("No Environment Provider Configs found.");
+      return;
+    }
+
+    PinotEnvironmentProviderFactory.init(environmentProviderConfigs);
+    String environmentProviderClassName = _serverConf.getProperty(Server.ENVIRONMENT_PROVIDER_CLASS_NAME);
+    if (environmentProviderClassName == null) {
+      LOGGER.info("No Class Name provided for Environment Provider.");
+      return;
+    }
+
+    // Fetch environment provider instance
+    PinotEnvironmentProvider pinotEnvironmentProvider = PinotEnvironmentProviderFactory.getEnvironmentProvider(
+        environmentProviderClassName.toLowerCase());
+
+    // Fetch the overridden pinot configuration containing custom configs.
+    Map<String, Object> overriddenPinotConfigs = pinotEnvironmentProvider.getEnvironment();
+
+    // Populate environment specific fields in the map.
+    String failureDomain = (String)overriddenPinotConfigs.get(PinotEnvironmentProvider.INSTANCE_FAILURE_DOMAIN);
+    if (failureDomain == null) {
+      LOGGER.info("No failure domain information found for instance: {}", _instanceId);
```

Comment:
Please add the class name that we used to get the fault domain info. That way, an operator will know which class to fix/update.